tests/test_dispatcher.py

```diff
@@ -174,6 +174,15 @@ def test_error_handler_that_raises_errors(self, dp):
 
         assert self.count == 1
 
+    def test_default_run_async(self, monkeypatch, dp):
+        def test_callback(*args, **kwargs):
+            self.received = "running async"
+
+        dp.add_handler(MessageHandler(Filters.all, test_callback))
```

Comment:
you need a different callback here. maybe just `lambda u, c: pass`. because the handler callback sets the flag, the test succeeds even though the default setting is not applied